Refactor parse_link_header

So we currently have `parse_link_header`

https://github.com/webcompat/webcompat.com/blob/b2b68c19881ea98c484a2c66e67db886ca6b02c3/webcompat/helpers.py#L342-L362

which is used at different places in the code, which returns a dictionary of links.

And what did I discovered today… cough cough… 🤦 

```python
>>> links = '<https://api.github.com/repositories/17914657/issues/1470/comments?page=2>; rel="next", <https://api.github.com/repositories/17914657/issues/1470/comments?page=2>; rel="last"' 
>>> import requests
>>> requests.utils.parse_header_links(links)
[{'url': 'https://api.github.com/repositories/17914657/issues/1470/comments?page=2', 'rel': 'next'}, {'url': 'https://api.github.com/repositories/17914657/issues/1470/comments?page=2', 'rel': 'last'}]
```

`parse_header_links` was added in 2012

That means removing also the test
https://github.com/webcompat/webcompat.com/blob/b2b68c19881ea98c484a2c66e67db886ca6b02c3/tests/unit/test_helpers.py#L146-L150

* [x] replace `parse_link_header` by `utils.parse_header_links`
* [x] remove `parse_link_header`
* [x] remove `test_parse_http_link_headers`
* [x] make sure all the tests are passing

There are some errors after I replaced `parse_link_header` by `utils.parse_header_links`
`(env)  ✘ apsychogirl@dell  ~/webcompat.com   3447/1 ●  pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.8.2, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/apsychogirl/webcompat.com, inifile: pytest.ini, testpaths: tests/unit
plugins: mock-3.2.0
collected 187 items                                                                                                                   

tests/unit/test_api_helpers.py ..                                                                                               [  1%]
tests/unit/test_api_urls.py F..........                                                                                         [  6%]
tests/unit/test_config.py ..                                                                                                    [  8%]
tests/unit/test_console_logs.py .....                                                                                           [ 10%]
tests/unit/test_form.py .....................                                                                                   [ 21%]
tests/unit/test_helpers.py .........................FFF                                                                         [ 36%]
tests/unit/test_http_caching.py ...                                                                                             [ 38%]
tests/unit/test_issues.py .........                                                                                             [ 43%]
tests/unit/test_rendering.py .....                                                                                              [ 45%]
tests/unit/test_template.py ............                                                                                        [ 52%]
tests/unit/test_tools_archive.py ............s.                                                                                 [ 59%]
tests/unit/test_tools_changelog.py ..                                                                                           [ 60%]
tests/unit/test_topsites.py ...                                                                                                 [ 62%]
tests/unit/test_uploads.py ...                                                                                                  [ 64%]
tests/unit/test_urls.py ................................                                                                        [ 81%]
tests/unit/test_webhook.py ....................                                                                                 [ 91%]
tests/unit/test_webhook_model.py ...............                                                                                [100%]

============================================================== FAILURES ===============================================================
___________________________________________ TestAPIURLs.test_api_comments_link_header_auth ____________________________________________

self = <unit.test_api_urls.TestAPIURLs testMethod=test_api_comments_link_header_auth>

    def test_api_comments_link_header_auth(self):
        """Pagination in comments Link response header.
    
        API access to comments greater than 30 returns HTTP Link
        in the response header.
        """
        with patch('webcompat.helpers.proxy_request') as github_data:
            # Create mock data. One api_response for each call.
            github_data.side_effect = [
                mock_api_response({
                    'status_code': 200,
                    'content': '[]',
                    'headers': {
                        'Link': '<https://api.github.com/repositories/17839063/issues/398/comments?page=2>; rel="next", <https://api.github.com/repositories/17839063/issues/398/comments?page=4>; rel="last"',  # noqa
                    },
                }),
                mock_api_response({'status_code': 200, 'content': '[]'})
            ]
>           rv = self.app.get('/api/issues/398/comments', environ_base=headers)

tests/unit/test_api_urls.py:123: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
env/lib/python3.8/site-packages/werkzeug/test.py:1006: in get
    return self.open(*args, **kw)
env/lib/python3.8/site-packages/flask/testing.py:222: in open
    return Client.open(
env/lib/python3.8/site-packages/werkzeug/test.py:970: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
env/lib/python3.8/site-packages/werkzeug/test.py:861: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
env/lib/python3.8/site-packages/werkzeug/test.py:1096: in run_wsgi_app
    app_rv = app(environ, start_response)
env/lib/python3.8/site-packages/flask/app.py:2464: in __call__
    return self.wsgi_app(environ, start_response)
env/lib/python3.8/site-packages/flask/app.py:2450: in wsgi_app
    response = self.handle_exception(e)
env/lib/python3.8/site-packages/flask/app.py:1867: in handle_exception
    reraise(exc_type, exc_value, tb)
env/lib/python3.8/site-packages/flask/_compat.py:39: in reraise
    raise value
env/lib/python3.8/site-packages/flask/app.py:2447: in wsgi_app
    response = self.full_dispatch_request()
env/lib/python3.8/site-packages/flask/app.py:1952: in full_dispatch_request
    rv = self.handle_user_exception(e)
env/lib/python3.8/site-packages/flask/app.py:1821: in handle_user_exception
    reraise(exc_type, exc_value, tb)
env/lib/python3.8/site-packages/flask/_compat.py:39: in reraise
    raise value
env/lib/python3.8/site-packages/flask/app.py:1950: in full_dispatch_request
    rv = self.dispatch_request()
env/lib/python3.8/site-packages/flask/app.py:1936: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
webcompat/api/endpoints.py:190: in proxy_comments
    comments_data = api_request('get', path, params=params,
webcompat/helpers.py:414: in wrapped_func
    return func(*args, **kwargs)
webcompat/helpers.py:478: in api_request
    get_response_headers(resource))
webcompat/helpers.py:205: in get_response_headers
    new_headers['link'] = rewrite_and_sanitize_link(headers.get('link'))
webcompat/helpers.py:339: in rewrite_and_sanitize_link
    return rewrite_links(sanitize_link(link_header))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues/398/comments?page=2>; rel="next", <https://api.github.com/repositories/17839063/issues/398/comments?page=4>; rel="last"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
_____________________________________________ TestHelpers.test_rewrite_and_sanitize_link ______________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_rewrite_and_sanitize_link>

    def test_rewrite_and_sanitize_link(self):
        """Rewrite and Sanitize Links test."""
        self.assertNotIn('access_token=',
>                        rewrite_and_sanitize_link(ACCESS_TOKEN_LINK))

tests/unit/test_helpers.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
webcompat/helpers.py:339: in rewrite_and_sanitize_link
    return rewrite_links(sanitize_link(link_header))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3&access_token=12345>; rel="next", <https://api...; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1&access_token=12345>; rel="prev"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
____________________________________________________ TestHelpers.test_rewrite_link ____________________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_rewrite_link>

    def test_rewrite_link(self):
        """Test we're correctly rewriting the passed in link."""
>       self.assertEqual(rewrite_links(GITHUB_ISSUES_LINK_HEADER),
                         REWRITTEN_ISSUES_LINK_HEADER)

tests/unit/test_helpers.py:85: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3>; rel="next", <https://api.github.com/reposit...per_page=50&page=1>; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1>; rel="prev"'

    def rewrite_links(link_header):
        """Rewrite Link header Github API endpoints to our own.
    
        <https://api.github.com/repositories/17839063/iss...&page=2>; rel="next",
        <https://api.github.com/repositories/17839063/iss...&page=4>; rel="last"
    
        is transformed into
    
        </api/issues?per_page=50&page=2>; rel="next",
        </api/issues?per_page=50&page=4>; rel="last" etc.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           uri = data['link']
E           KeyError: 'link'

webcompat/helpers.py:296: KeyError
___________________________________________________ TestHelpers.test_sanitize_link ____________________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_sanitize_link>

    def test_sanitize_link(self):
        """Test that we're removing access_token parameters."""
>       self.assertNotIn('access_token=', sanitize_link(ACCESS_TOKEN_LINK))

tests/unit/test_helpers.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3&access_token=12345>; rel="next", <https://api...; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1&access_token=12345>; rel="prev"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
======================================================= short test summary info =======================================================
FAILED tests/unit/test_api_urls.py::TestAPIURLs::test_api_comments_link_header_auth - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_rewrite_and_sanitize_link - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_rewrite_link - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_sanitize_link - KeyError: 'link'
============================================== 4 failed, 182 passed, 1 skipped in 2.18s ===============================================
`

There are some errors after I replaced `parse_link_header` by `utils.parse_header_links`
```bash
(env)  ✘ apsychogirl@dell  ~/webcompat.com   3447/1 ●  pytest
========================================================= test session starts =========================================================
platform linux -- Python 3.8.2, pytest-5.4.3, py-1.9.0, pluggy-0.13.1
rootdir: /home/apsychogirl/webcompat.com, inifile: pytest.ini, testpaths: tests/unit
plugins: mock-3.2.0
collected 187 items                                                                                                                   

tests/unit/test_api_helpers.py ..                                                                                               [  1%]
tests/unit/test_api_urls.py F..........                                                                                         [  6%]
tests/unit/test_config.py ..                                                                                                    [  8%]
tests/unit/test_console_logs.py .....                                                                                           [ 10%]
tests/unit/test_form.py .....................                                                                                   [ 21%]
tests/unit/test_helpers.py .........................FFF                                                                         [ 36%]
tests/unit/test_http_caching.py ...                                                                                             [ 38%]
tests/unit/test_issues.py .........                                                                                             [ 43%]
tests/unit/test_rendering.py .....                                                                                              [ 45%]
tests/unit/test_template.py ............                                                                                        [ 52%]
tests/unit/test_tools_archive.py ............s.                                                                                 [ 59%]
tests/unit/test_tools_changelog.py ..                                                                                           [ 60%]
tests/unit/test_topsites.py ...                                                                                                 [ 62%]
tests/unit/test_uploads.py ...                                                                                                  [ 64%]
tests/unit/test_urls.py ................................                                                                        [ 81%]
tests/unit/test_webhook.py ....................                                                                                 [ 91%]
tests/unit/test_webhook_model.py ...............                                                                                [100%]

============================================================== FAILURES ===============================================================
___________________________________________ TestAPIURLs.test_api_comments_link_header_auth ____________________________________________

self = <unit.test_api_urls.TestAPIURLs testMethod=test_api_comments_link_header_auth>

    def test_api_comments_link_header_auth(self):
        """Pagination in comments Link response header.
    
        API access to comments greater than 30 returns HTTP Link
        in the response header.
        """
        with patch('webcompat.helpers.proxy_request') as github_data:
            # Create mock data. One api_response for each call.
            github_data.side_effect = [
                mock_api_response({
                    'status_code': 200,
                    'content': '[]',
                    'headers': {
                        'Link': '<https://api.github.com/repositories/17839063/issues/398/comments?page=2>; rel="next", <https://api.github.com/repositories/17839063/issues/398/comments?page=4>; rel="last"',  # noqa
                    },
                }),
                mock_api_response({'status_code': 200, 'content': '[]'})
            ]
>           rv = self.app.get('/api/issues/398/comments', environ_base=headers)

tests/unit/test_api_urls.py:123: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
env/lib/python3.8/site-packages/werkzeug/test.py:1006: in get
    return self.open(*args, **kw)
env/lib/python3.8/site-packages/flask/testing.py:222: in open
    return Client.open(
env/lib/python3.8/site-packages/werkzeug/test.py:970: in open
    response = self.run_wsgi_app(environ.copy(), buffered=buffered)
env/lib/python3.8/site-packages/werkzeug/test.py:861: in run_wsgi_app
    rv = run_wsgi_app(self.application, environ, buffered=buffered)
env/lib/python3.8/site-packages/werkzeug/test.py:1096: in run_wsgi_app
    app_rv = app(environ, start_response)
env/lib/python3.8/site-packages/flask/app.py:2464: in __call__
    return self.wsgi_app(environ, start_response)
env/lib/python3.8/site-packages/flask/app.py:2450: in wsgi_app
    response = self.handle_exception(e)
env/lib/python3.8/site-packages/flask/app.py:1867: in handle_exception
    reraise(exc_type, exc_value, tb)
env/lib/python3.8/site-packages/flask/_compat.py:39: in reraise
    raise value
env/lib/python3.8/site-packages/flask/app.py:2447: in wsgi_app
    response = self.full_dispatch_request()
env/lib/python3.8/site-packages/flask/app.py:1952: in full_dispatch_request
    rv = self.handle_user_exception(e)
env/lib/python3.8/site-packages/flask/app.py:1821: in handle_user_exception
    reraise(exc_type, exc_value, tb)
env/lib/python3.8/site-packages/flask/_compat.py:39: in reraise
    raise value
env/lib/python3.8/site-packages/flask/app.py:1950: in full_dispatch_request
    rv = self.dispatch_request()
env/lib/python3.8/site-packages/flask/app.py:1936: in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
webcompat/api/endpoints.py:190: in proxy_comments
    comments_data = api_request('get', path, params=params,
webcompat/helpers.py:414: in wrapped_func
    return func(*args, **kwargs)
webcompat/helpers.py:478: in api_request
    get_response_headers(resource))
webcompat/helpers.py:205: in get_response_headers
    new_headers['link'] = rewrite_and_sanitize_link(headers.get('link'))
webcompat/helpers.py:339: in rewrite_and_sanitize_link
    return rewrite_links(sanitize_link(link_header))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues/398/comments?page=2>; rel="next", <https://api.github.com/repositories/17839063/issues/398/comments?page=4>; rel="last"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
_____________________________________________ TestHelpers.test_rewrite_and_sanitize_link ______________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_rewrite_and_sanitize_link>

    def test_rewrite_and_sanitize_link(self):
        """Rewrite and Sanitize Links test."""
        self.assertNotIn('access_token=',
>                        rewrite_and_sanitize_link(ACCESS_TOKEN_LINK))

tests/unit/test_helpers.py:99: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _
webcompat/helpers.py:339: in rewrite_and_sanitize_link
    return rewrite_links(sanitize_link(link_header))
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3&access_token=12345>; rel="next", <https://api...; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1&access_token=12345>; rel="prev"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
____________________________________________________ TestHelpers.test_rewrite_link ____________________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_rewrite_link>

    def test_rewrite_link(self):
        """Test we're correctly rewriting the passed in link."""
>       self.assertEqual(rewrite_links(GITHUB_ISSUES_LINK_HEADER),
                         REWRITTEN_ISSUES_LINK_HEADER)

tests/unit/test_helpers.py:85: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3>; rel="next", <https://api.github.com/reposit...per_page=50&page=1>; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1>; rel="prev"'

    def rewrite_links(link_header):
        """Rewrite Link header Github API endpoints to our own.
    
        <https://api.github.com/repositories/17839063/iss...&page=2>; rel="next",
        <https://api.github.com/repositories/17839063/iss...&page=4>; rel="last"
    
        is transformed into
    
        </api/issues?per_page=50&page=2>; rel="next",
        </api/issues?per_page=50&page=4>; rel="last" etc.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           uri = data['link']
E           KeyError: 'link'

webcompat/helpers.py:296: KeyError
___________________________________________________ TestHelpers.test_sanitize_link ____________________________________________________

self = <unit.test_helpers.TestHelpers testMethod=test_sanitize_link>

    def test_sanitize_link(self):
        """Test that we're removing access_token parameters."""
>       self.assertNotIn('access_token=', sanitize_link(ACCESS_TOKEN_LINK))

tests/unit/test_helpers.py:94: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

link_header = '<https://api.github.com/repositories/17839063/issues?per_page=50&page=3&access_token=12345>; rel="next", <https://api...; rel="first", <https://api.github.com/repositories/17839063/issues?per_page=50&page=1&access_token=12345>; rel="prev"'

    def sanitize_link(link_header):
        """Remove any oauth tokens from the Link header from GitHub.
    
        see Also rewrite_links.
        """
        header_link_data = requests.utils.parse_header_links(link_header)
        for data in header_link_data:
>           data['link'] = remove_oauth(data['link'])
E           KeyError: 'link'

webcompat/helpers.py:318: KeyError
======================================================= short test summary info =======================================================
FAILED tests/unit/test_api_urls.py::TestAPIURLs::test_api_comments_link_header_auth - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_rewrite_and_sanitize_link - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_rewrite_link - KeyError: 'link'
FAILED tests/unit/test_helpers.py::TestHelpers::test_sanitize_link - KeyError: 'link'
============================================== 4 failed, 182 passed, 1 skipped in 2.18s ===============================================
(env)  ✘ apsychogirl@dell  ~/webcompat.com   3447/1 ●  


```

--- a/tests/unit/test_helpers.py
+++ b/tests/unit/test_helpers.py
@@ -31,7 +31,6 @@ from webcompat.helpers import get_serialized_value
 from webcompat.helpers import get_version_string
 from webcompat.helpers import is_json_object
 from webcompat.helpers import normalize_api_params
-from webcompat.helpers import parse_link_header
 from webcompat.helpers import prepare_form
 from webcompat.helpers import rewrite_and_sanitize_link
 from webcompat.helpers import rewrite_links
@@ -48,7 +47,7 @@ GITHUB_SEARCH_LINK_HEADER = '<https://api.github.com/search/issues?q=taco&page=2
 REWRITTEN_SEARCH_LINK_HEADER = '</api/issues/search?q=taco&page=2>; rel="next", </api/issues/search?q=taco&page=26>; rel="last"'  # noqa
 GITHUB_COMMENTS_LINK_HEADER = '<https://api.github.com/repositories/17839063/issues/398/comments?page=2>; rel="next", <https://api.github.com/repositories/17839063/issues/398/comments?page=4>; rel="last"'  # noqa
 REWRITTEN_COMMENTS_LINK_HEADER = '</api/issues/398/comments?page=2>; rel="next", </api/issues/398/comments?page=4>; rel="last"'  # noqa
-PARSED_LINKED_HEADERS = [{'link': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=3', 'rel': 'next'}, {'link': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=4', 'rel': 'last'}, {'link': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=1', 'rel': 'first'}, {'link': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=1', 'rel': 'prev'}]  # noqa
+PARSED_LINKED_HEADERS = [{'url': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=3', 'rel': 'next'}, {'url': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=4', 'rel': 'last'}, {'url': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=1', 'rel': 'first'}, {'url': 'https://api.github.com/repositories/17839063/issues?per_page=50&page=1', 'rel': 'prev'}]  # noqa
 FIREFOX_UA = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10.11; rv:48.0) Gecko/20100101 Firefox/48.0'  # noqa
 FIREFOX_MOBILE_UA_OLD = 'Mozilla/5.0 (Android; Mobile; rv:40.0) Gecko/40.0 Firefox/40.0'  # noqa
 FIREFOX_MOBILE_UA = 'Mozilla/5.0 (Android 6.0.1; Mobile; rv:40.0) Gecko/40.0 Firefox/40.0'  # noqa
@@ -142,12 +141,6 @@ class TestHelpers(unittest.TestCase):
         self.assertEqual({'order': 'desc', 'foo': 'bar'},
                          normalize_api_params({'foo': 'bar',
                                               'direction': 'desc'}))
-
-    def test_parse_http_link_headers(self):
-        """Test HTTP Links parsing for GitHub only."""
-        link_header = GITHUB_ISSUES_LINK_HEADER
-        parsed_headers = PARSED_LINKED_HEADERS
-        self.assertEqual(parse_link_header(link_header), parsed_headers)
 
     def test_format_http_link_headers(self):
         """Test HTTP Links formating."""

--- a/webcompat/helpers.py
+++ b/webcompat/helpers.py
@@ -26,6 +26,7 @@ from flask import make_response
 from flask import request
 from flask import session
 import requests
+from requests.utils import parse_header_links
 from ua_parser import user_agent_parser
 
 from webcompat import app
@@ -291,9 +292,9 @@ def rewrite_links(link_header):
     </api/issues?per_page=50&page=2>; rel="next",
     </api/issues?per_page=50&page=4>; rel="last" etc.
     """
-    header_link_data = parse_link_header(link_header)
+    header_link_data = parse_header_links(link_header)
     for data in header_link_data:
-        uri = data['link']
+        uri = data['url']
         uri_tuple = urllib.parse.urlsplit(uri)
         path = uri_tuple.path
         query = uri_tuple.query
@@ -304,7 +305,7 @@ def rewrite_links(link_header):
         elif path.startswith('/search/issues'):
             path = 'issues/search'
         api_path = '{}{}'.format('/api/', path)
-        data['link'] = urllib.parse.urlunsplit(('', '', api_path, query, ''))
+        data['url'] = urllib.parse.urlunsplit(('', '', api_path, query, ''))
     return format_link_header(header_link_data)
 
 
@@ -313,9 +314,9 @@ def sanitize_link(link_header):
 
     see Also rewrite_links.
     """
-    header_link_data = parse_link_header(link_header)
+    header_link_data = parse_header_links(link_header)
     for data in header_link_data:
-        data['link'] = remove_oauth(data['link'])
+        data['url'] = remove_oauth(data['url'])
     return format_link_header(header_link_data)
 
 
@@ -339,32 +340,9 @@ def rewrite_and_sanitize_link(link_header):
     return rewrite_links(sanitize_link(link_header))
 
 
-def parse_link_header(link_header):
-    """Return a structured list of objects for an HTTP Link header.
-
-    This is adjusted for github links it will break in a more generic case.
-    Do not use this code for your own HTTP Link header parsing.
-    Use something like https://pypi.python.org/pypi/LinkHeader/ instead.
-    """
-    links_list = link_header.split(',')
-    header_link_data = []
-    for link in links_list:
-        # Assuming that link is `<uri>; rel="blah"`. Github only.
-        uri_info, rel_info = link.split(';')
-        uri_info = uri_info.strip()
-        rel_info = rel_info.strip()
-        rel_keyword, value = rel_info.split('=')
-        # rel values have the form `rel="foo"`, we want `foo`.
-        rel_value = value[1:-1]
-        # uri have the form `<http://…>`, we want `http://…`.
-        uri = uri_info[1:-1]
-        header_link_data.append({'link': uri, 'rel': rel_value})
-    return header_link_data
-
-
 def format_link_header(link_header_data):
     """Return a string ready to be used in a Link: header."""
-    links = ['<{0}>; rel="{1}"'.format(data['link'], data['rel'])
+    links = ['<{0}>; rel="{1}"'.format(data['url'], data['rel'])
              for data in link_header_data]
     return ', '.join(links)
 

